filer/models/filemodels.py

```diff
@@ -81,6 +81,35 @@ def matches_file_type(cls, iname, ifile, request):
     def __init__(self, *args, **kwargs):
         super(File, self).__init__(*args, **kwargs)
         self._old_is_public = self.is_public
+        self.file_data_changed(post_init=True)
+
+    def file_data_changed(self, post_init=False):
+        """
+        This is called whenever self.file changes (including initial set in __init__).
+        MultiStorageFileField has a custom descriptor which calls this function when
+        field value is changed.
+        Returns True if data related attributes were updated, False otherwise.
+        """
+        if hasattr(self, '_file_data_changed_hint'):
```

Comment:
Why not using ``True``/``False``/``None`` instead of adding and removing the attribute?